src/nvim/misc1.c

```diff
@@ -1512,13 +1500,14 @@ void ins_char_bytes(char_u *buf, int charlen)
   memmove(p + newlen, oldp + col + oldlen,
       (size_t)(linelen - col - oldlen));
 
-  /* Insert or overwrite the new character. */
-  memmove(p, buf, charlen);
-  i = charlen;
+  // Insert or overwrite the new character.
+  assert(charlen >= 0);
```

Comment:
Alternatively, we can change `charlen` to `size_t`, to move the conversion to the callsites, all where the non-negativeness is clear. (and presumably `newlen` as well)